GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2397,7 +2397,7 @@ private void generateListOfChangesInSubmodulesChangesToolStripMenuItem_Click(obj
             }
 
             Dictionary<string, string> modules = stagedFiles
-                .Where(item => item.IsSubmodule)
+                .Where(item => item.IsSubmodule && Directory.Exists(_fullPathResolver.Resolve(item.Name)))
```

Comment:
Depending on feedback to #8315 this should maybe be written as 
```
.Where(item => item.IsSubmodule && item.Name != null && Directory.Exists(_fullPathResolver.Resolve(item.Name)))
```
But I see no reason for Name to be null here

Reply:
Another change was needed, pushed.